### DISCUSSION TITLE
Teensy 4.1 how to load or change settings

### DISCUSSION THREAD

**Discussion by Cncninja** [terjeio/grblHAL#158] [Q&A]
Hello and thanks for this great project, I bought a @phil-barrett Breakout and loaded the default 4 axes from his repo, but I notice there are a lot more features in I/O sender if I select a dummy comm port, such as tool table, trimatic driver tunning, etc, but when I connect to the real Teensy these options are not there. where can I add this?

Thanks

**Comment by terjeio** [comment]
grblHAL supports many different cards and config options. ioSender configures itself on connection to reflect that - based on data reported from the controller.

Trinamic tuning requires a card that has Trinamic support. I have been working on that recently and have added initial support for TMC2130 for Bigtreetech SKR 1.x (LPC1768) and for the next iteration of my [Nucleo-64 breakout](https://github.com/terjeio/CNC_Breakout_Nucleo64) (STM32F4xx). My [CNC BoosterPack](https://github.com/terjeio/CNC_Boosterpack) board also has support for this driver (for TI boards/drivers such as MSP432). Currently no boards exists for Teensy 4.x that supports Trinamic drivers (except a prototype I have made for 4.0 that allows use of my CNC BoosterPack).

Some options are configurable - the tooltable is one that does not need any driver support code, this can be enabled in the source and compiled in:

https://github.com/terjeio/grblHAL/blob/2569bb233f63f109d2217c0a56b624182af6d9ad/grbl/config.h#L437-L440

Lathe mode (run time configurable via setting) and lathe spindle sync are other options supported by ioSender.

So some options are available for the Teensy, some not.

**Reply by Cncninja** [reply]
Thank you, ill dive, into how to compile, You menched Grblhal support for the Bigtreetech SKR 1.x (LPC1768), where might I find the info on this? the boards are cheap and more than enough for my application.

**Reply by terjeio** [reply]
>  You menched Grblhal support for the Bigtreetech SKR 1.x (LPC1768), where might I find the info on this? the boards are cheap and mo

[Here?](https://github.com/terjeio/grblHAL/tree/test/drivers/LPC1769) I have not added a link to the 1.3 board yet.

**Comment by Cncninja** [comment]
Thank you very much do you feel the SKR 1.4 turbo is a good option, I just
ran IO sender for 3 hours and it works flawlessly

Thank you for your efforts do you have a support link?
[image: IMG_1338.jpg][image: IMG_1339.jpg]

**Reply by terjeio** [reply]
I do not have opinions about this board as I do not own one myself. However, there are some negative comments about it if you do a search...

You may also want to check the open issues at Bigtreetechs [github repository](https://github.com/bigtreetech/BIGTREETECH-SKR-V1.3). 

> Thank you for your efforts do you have a support link?

Images does not show. Support link for what?

**Comment by Cncninja** [comment]
support link to donate
